### DISCUSSION TITLE
Issue with displaying Geoip coordinates on Kibana Maps

### DISCUSSION THREAD

**Discussion by esbat-max** [Security-Onion-Solutions/securityonion#6240] [Unsupported Versions]
Hello,

I am using security onion release 2.3.70 in standalone mode.

I am collecting OWA logs via filebeat to security onion. Dataflow is as follows: Beats -> Elasticsearch -> Index Template -> Ingest Processor -> Index.

Using the geoip processor pipeline I converted the device ip field into geoip coordinates. However, when I try to put the coordinates on maps, the result is empty map.

Any advice of missing configuration ?

thanks,
El

**Comment by weslambert** [comment]
What do the mappings for the geo fields look like if you go to Kibana, then `Stack Management` -> `Index Patterns` -> `*:so-*`? 

Do you see the geoip fields present?

 If so, how are they typed (data type)?

**Reply by esbat-max** [reply]
Hello Wes,

Actually the above case is solved by using the set processor to take the values for LAT and LON and assign both values to a geo.location field of type geo_point. The IPs are distributed successfully on maps, however, if I use Cisco talos (or any other open source) database I am receiving erroneous data. i.e: IP regsitered in UK in geo.conf and showing in Lebanon if using Cisco Talos.

I think the geoip.conf requires immediate update.

Thanks,
E

**Reply by weslambert** [reply]
To be honest, it's been a while since I've looked at this -- I'd have to look back and see if there is anything we can do with the Geo IP DB. IIRC it's the default DB distributed by Elastic. As far as the geo_point fields, I'll have to look into it some more.

**Reply by ccamejo1** [reply]
Wes,

We were having the same issue mapping geo points in Kibana maps.  After trying different workarounds some more successful than others, I believe we narrowed down the source of the problem.

In the elasticsearch templates field mappings:
/opt/so/saltstack/default/salt/elasticsearch/templates/so/so-common-template.json.jinja 
There are 2 fields typed as 'geo_point' (source_geo.location, destination_geo.location) but there is never values set to these fields.  

During ingestion the geoip processor that performs the lookups in  opt/so/saltstack/default/salt/elasticsearch/files/ingest-dynamic/common assigns the 'location' field to source.geo.location and destination.geo.location as seen below:
{ 	
	"geoip":  { 
		"field": "destination.ip",
		"target_field": "destination.geo",
		"database_file": "GeoLite2-City.mmdb",
		"ignore_missing": true,
		"ignore_failure": true,
		"properties": ["ip", "country_iso_code", "country_name", "continent_name", "region_iso_code", "region_name", "city_name", "timezone", "location"]
	}
    },
    { 
	"geoip":  {
		"field": "source.ip",
		"target_field": "source.geo",
		"database_file": "GeoLite2-City.mmdb",
		"ignore_missing": true,
		"ignore_failure": true,
		"properties": ["ip", "country_iso_code", "country_name", "continent_name", "region_iso_code", "region_name", "city_name", "timezone", "location"]
	}
    },

The source_geo and destination_geo is used for the ASN enrichment, but there is no 'location' field added here:

    {
        "geoip":  {
                "field": "destination.ip",
                "target_field": "destination_geo",
                "database_file": "GeoLite2-ASN.mmdb",
                "ignore_missing": true,
                "ignore_failure": true,
                "properties": ["ip", "asn", "organization_name", "network"]
        }
    },
    {
        "geoip":  {
                "field": "source.ip",
                "target_field": "source_geo",
                "database_file": "GeoLite2-ASN.mmdb",
                "ignore_missing": true,
                "ignore_failure": true,
                "properties": ["ip", "asn", "organization_name", "network"]
        }

This can be solved as suggested by esbat-max by using a set processor to assign the fields [source/destination].geo.location.lat and [source/destination].geo.location.lon to [source/destination]_geo.location, but that adds additional cycles to every record passing through the pipeline.  While this may be acceptable for smaller deployments, when ingesting hundreds of millions events per hour every millisecond counts.

I believe changing the mapping from [source/destination]_geo to [source/destination].geo for future new releases being used for new installs will fix the root of the problem.  

**Changing this value on existing deployments will cause a field type conflict as long as there is data that exists in indices that have dynamically set the [source/destination].geo.location as object instead of geo_point.  Perhaps a combination of using set as esbat-max suggested to get it working with [source/destination]_geo location while the [source/destination].geo.location as geo_point makes it through the indices and the conflict self resolves?

Thanks,

C²

**Reply by esbat-max** [reply]
Hello ccamejo1,

Let me describe the steps I followed that makes the geo ip operational for my setup. I am using SO in standalone mode (single node):

Note: Traffic flows: Filebeat -> ingest pipelines -> Elastic database <-> Kibana

1. First, I have converted the public IP field from text to IP using the geoip processor. Automatically additional fields are added to the discover page (i.e latitude, longitude, city name, country name, etc.).
2. Second, using the set processor, in the field textbox choose an already defined field of geo_point type. You can find the available fields type under kibana index pattern.
3. Third, The tricky part here, at least for me, in the value textbox of the set processor you have to use double curly braces in order for the set processor to read the value of the fields. i.e: {{geo.location.lat}},{{geo.location.lon}} in my case.

Then, you go to maps:
1. Choose datasource: Documents
2. Index pattern: filebeat or any other containing the lat lon values of the public IP
3. Geo.spatial field being the field of geo_point type (predefined in the your index as mentioned above).

Finally, all public IPs location will be automatically populated to your map.

This was my solution, hopefully, It is helpful for you and I did not miss any step.


Thanks,
E

**Reply by eric-holtzclaw** [reply]
I just tried 2.3 GA and I don't see any GeoIp in Kibana.   I do see in the Index Pattern *:so-* and says searchable
